ckanext/dbca/logic/validators.py

```diff
@@ -55,9 +56,70 @@ def dbca_validate_geojson(value):
     return value
 
 
+def dbca_resource_size(key, data, errors, context):
+    # The new validator checks to see if the resource is an upload
+    url_type_key = key[0:2] + ('url_type',)
+    resource_upload = data.get(url_type_key) == 'upload'
+    if not resource_upload:
+        return
+
+    # Get the max resource size from the CKAN config
+    max_resource_size = int(tk.config.get('ckan.max_resource_size'))
+    sysadmin_resource_upload_limit = int(tk.config.get('ckanext.dbca.sysadmin_resource_upload_limit'))
+    org_admin_resource_upload_limit = int(tk.config.get('ckanext.dbca.org_admin_resource_upload_limit'))
+    org_editor_resource_upload_limit = int(tk.config.get('ckanext.dbca.org_editor_resource_upload_limit'))
+
+    # Convert the max resource size from MB to bytes
+    resource_size_limit_bytes = max_resource_size * 1024 * 1024
+    sysadmin_resource_upload_limit_bytes = sysadmin_resource_upload_limit * 1024 * 1024
+    org_admin_resource_upload_limit_bytes = org_admin_resource_upload_limit * 1024 * 1024
+    org_editor_resource_upload_limit_bytes = org_editor_resource_upload_limit * 1024 * 1024
+
+    # Get the resource size from the data, and throw error if it's over the system limit.
```

Comment:
@awset I am not sure if we need this check?
I added it just for the POC Is it still required?
This logic is handled in the CKAN uploader https://github.com/ckan/ckan/blob/2.10/ckan/lib/uploader.py#L41